tests/rptest/tests/datalake/schema_evolution_test.py

```diff
@@ -0,0 +1,207 @@
+        {
+            "name": "verifier_string",
+            "type": "string"
+        },
+        {
+            "name": "ordinal",
+            "type": "int"
+        }
+    ]
+}
+"""
+
+legal_evo_schema_avro = """
+{
+    "type": "record",
+    "name": "VerifierRecord",
+    "fields": [
+        {
+            "name": "verifier_string",
+            "type": "string"
+        },
+        {
+            "name": "ordinal",
+            "type": "long"
+        }
+    ]
+}
+"""
+
+illegal_evo_schema_avro = """
+{
+    "type": "record",
+    "name": "VerifierRecord",
+    "fields": [
+        {
+            "name": "verifier_string",
+            "type": "double"
+        },
+        {
+            "name": "ordinal",
+            "type": "long"
+        }
+    ]
+}
+"""
+
+
+class SchemaEvolutionTest(RedpandaTest):
+    def __init__(self, test_context):
+        super(SchemaEvolutionTest, self).__init__(
+            test_context=test_context,
+            num_brokers=1,
+            si_settings=SISettings(test_context,
+                                   cloud_storage_enable_remote_read=False,
+                                   cloud_storage_enable_remote_write=False),
+            extra_rp_conf={
+                "iceberg_enabled": True,
+                "iceberg_catalog_commit_interval_ms": 5000,
+            },
+            schema_registry_config=SchemaRegistryConfig())
+
+    def avro_stream_config(self, topic, subject, mapping: dict = {}):
+        return {
+            "input": {
+                "generate": {
+                    "mapping": "root = counter()",
+                    "interval": "",
+                    "count": 1000,
+                    "batch_size": 1
+                }
+            },
+            "pipeline": {
+                "processors": [{
+                    "mapping":
+                    f"""
+                            root.verifier_string = {mapping.get('verifier_string', 'uuid_v4()')}
+                            root.ordinal = {mapping.get('ordinal', 'this')}
+                    """
+                }, {
+                    "schema_registry_encode": {
+                        "url": self.redpanda.schema_reg().split(",")[0],
+                        "subject": subject,
+                        "refresh_period": "10s"
+                    }
+                }]
+            },
+            "output": {
+                "redpanda": {
+                    "seed_brokers": self.redpanda.brokers_list(),
+                    "topic": topic,
+                }
+            }
+        }
+
+    def setUp(self):
+        pass
+
+    def _create_schema(self, subject: str, schema: str, schema_type="avro"):
+        rpk = RpkTool(self.redpanda)
+        with tempfile.NamedTemporaryFile(suffix=f".{schema_type}") as tf:
+            tf.write(bytes(schema, 'UTF-8'))
+            tf.flush()
+            rpk.create_schema(subject, tf.name)
+
+    @cluster(num_nodes=4, log_allow_list=["Schema \d+ already exists"])
```

Comment:
> `log_allow_list...`

@andrwng - this occurs on maybe 5% of runs. workers racing on the in-memory metadata or something like that? seems benign with respect to this PR, but flagging here in case it's surprising to you.